A86-xds-http-connect.md

```diff
@@ -0,0 +1,108 @@
+A86: xDS-Based HTTP CONNECT
+----
+* Author(s): @markdroth
+* Approver: @ejona86, @dfawley
+* Status: {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2024-10-02
+* Discussion at: https://groups.google.com/g/grpc-io/c/WiqQ7h003fE
+
+## Abstract
+
+This document provides a design for gRPC to support configuring HTTP
+CONNECT functionality via xDS.
+
+## Background
+
+gRPC currently supports HTTP CONNECT proxies, as described in [gRFC A1].
+gRPC also supports obtaining configuration via xDS, as originally
+described in [gRFC A27].  This document describes new functionality to
+allow configuring use of an HTTP CONNECT proxy via xDS.
+
+### Related Proposals: 
+* [gRFC A1: HTTP CONNECT Proxy Support][gRFC A1]
+* [gRFC A27: xDS-Based Global Load Balancing][gRFC A27]
+* [gRFC A29: xDS-Based Security for gRPC Clients and Servers][gRFC A29]
+* [gRFC A83: A83: xDS GCP Authentication Filter][gRFC A83]
+
+[gRFC A1]: A1-http-connect-proxy-support.md
+[gRFC A27]: A27-xds-global-load-balancing.md
+[gRFC A29]: A29-xds-tls-security.md
+[gRFC A83]: A83-xds-gcp-authn-filter.md
+
+## Proposal
+
+The xDS configuration for HTTP CONNECT is done via the
+[`Http11ProxyUpstreamTransport`](https://github.com/envoyproxy/envoy/blob/d6120f3c769e70c988ddcc5c7e9cbc2737b5f63c/api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto#L36)
+transport socket wrapper.  This wrapper is sent in the CDS [`transport_socket`
+field](https://github.com/envoyproxy/envoy/blob/d6120f3c769e70c988ddcc5c7e9cbc2737b5f63c/api/envoy/config/cluster/v3/cluster.proto#L1099),
+as a wrapper around whatever transport socket is actually used.
+
+Note that the only transport socket implementation that gRPC currently
+supports is `UpstreamTlsContext`, which is used to configure TLS
+security, as per [gRFC A29].  We honor the xDS TLS configuration only if
+the application uses `XdsCredentials` when creating the gRPC channel.
+With this design, we will now honor the `Http11ProxyUpstreamTransport`
+transport socket wrapper regardless of whether `XdsCredentials` is used.
+However, we will treat the underlying transport socket wrapped by the
+`Http11ProxyUpstreamTransport` wrapper exactly the way that we would
+previously have treated the transport socket specified directly in the CDS
+resource: if the underlying transport socket is `UpstreamTlsContext`, then
+we will honor it only if `XdsCredentials` is used, and if it is any other
```

Comment:
So it won't behave the same, because `Http11ProxyUpstreamTransport` is allowed as the top-level.

Reply:
That's correct.  I've attempted to clarify the wording here.